File: python/test_performance_primes.py
##
# 
# Ganesh Bhat
# https://github.com/ganeshkbhat/fastprimecalculations
# Fastest method of Prime Number Calculations
# Functions and Alternate ways for calculation of Prime Numbers
# 
# ** prime(n) ** is the Recommended Way for calculation checks of prime numbers
# All others are alternative ways
#
##

from primes import *
from primes_alternate_ways import *
import time
import math


def tests_performance_isPrimeConventionalWayArr():
    isPrimeConventionalWaySum = 0
    for i in range(1, 1000000):
        start = time.perf_counter()
        isPrimeConventionalWay(30000239)
        end = time.perf_counter()
        isPrimeConventionalWaySum += (end - start)
    return isPrimeConventionalWaySum

print( "isPrimeConventionalWayArr Sum: ", tests_performance_isPrimeConventionalWayArr() )


def tests_performance_isPrimeSquarerootWayArr():
    isPrimeSquarerootWaySum = 0
    for i in range(1, 1000000):
        start = time.perf_counter()
        isPrimeSquarerootWay(30000239)
        end = time.perf_counter()
        isPrimeSquarerootWaySum += (end - start)
    return isPrimeSquarerootWaySum

print( "isPrimeSquarerootWayArr Sum: ", tests_performance_isPrimeSquarerootWayArr() )


def tests_performance_isprimeAKSWayArr():
    isprimeAKSWaySum = 0
    for i in range(1, 1000000):
        start = time.perf_counter()
        isprimeAKSWay(30000239)
        end = time.perf_counter()
        isprimeAKSWaySum += (end - start)
    return isprimeAKSWaySum

print( "isprimeAKSWayArr Sum: ", tests_performance_isprimeAKSWayArr() )


def tests_performance_primeArr():
    primeSum = 0
    for i in range(1, 1000000):
        start = time.perf_counter()
        prime(30000239)
        end = time.perf_counter()
        primeSum += (end - start)
    return primeSum

print( "primeArr Sum: ", tests_performance_primeArr() )


# # # PRINT RESULTS OF PERFORMANCE TO FILES
# # #
# # with open("./isPrimeConventionalWayFile.log", 'w') as f:
# #     for item in isPrimeConventionalWayArr:
# #         f.write("%s\n" % item)
# # with open("./isPrimeSquarerootWayFile.log", 'w') as f:
# #     for item in isPrimeSquarerootWayArr:
# #         f.write("%s\n" % item)
# # with open("./primeFile.log", 'w') as f:
# #     for item in primeArr:
# #         f.write("%s\n" % item)
# # with open("./isprimeAKSWayFile.log", 'w') as f:
# #     for item in isprimeAKSWayArr:
# #         f.write("%s\n" % item)

File: php/test_performance_primes.php
<?php
/**
 * Ganesh Bhat
 * https://github.com/ganeshkbhat/fastprimecalculations
 * Fastest method of Prime Number Calculations
 * Tests for Prime Numbers
*/

require_once("./primes.php");
require_once("./primes_alternate_ways.php");


$iterations = 10000000;

// For Average (removing this due to memory related issues in php during calculation of 10 Million records)
// Average calculation is being used for all other languages

function calculateAverage($arr) {
    return array_sum($arr) / count($arr);
}

function tests_performance_isPrimeConventionalWayArr(){
    global $iterations;
    global $isPrimeConventionalWay;
    $isPrimeConventionalWayArr = [];
    $isPrimeConventionalWaySum = 0;
    for ($i = 1; $i <= $iterations; $i++){
        $start = microtime(true);
        isPrimeConventionalWay(30000239);
        $end = microtime(true);
        // array_push($isPrimeConventionalWayArr, $time);
        $isPrimeConventionalWaySum = $isPrimeConventionalWaySum + ($end - $start);
    }
    return $isPrimeConventionalWaySum;
}

// $result = calculateAverage($isPrimeConventionalWayArr);
// echo "isPrimeConventionalWayArr: ", $result, "\n";
// $isPrimeConventionalWayArr = [];
echo "isPrimeConventionalWaySum: ", tests_performance_isPrimeConventionalWayArr(), "\n";
clearstatcache();


function tests_performance_isPrimeSquarerootWayArr(){
    global $iterations;
    global $isPrimeSquarerootWay;
    $isPrimeSquarerootWayArr = [];
    $isPrimeSquarerootWaySum = 0;
    for ($i = 1; $i <= $iterations; $i++){
        $start = microtime(true);
        isPrimeSquarerootWay(30000239);
        $end = microtime(true);
        // array_push($isPrimeSquarerootWayArr, $time);
        $isPrimeSquarerootWaySum = $isPrimeSquarerootWaySum + ($end - $start);
    } 
    return $isPrimeSquarerootWaySum;
}

// $result = calculateAverage($isPrimeSquarerootWayArr);
// echo "isPrimeSquarerootWayArr: ", $result, "\n";
// $isPrimeSquarerootWayArr = [];
echo "isPrimeSquarerootWaySum: ", tests_performance_isPrimeSquarerootWayArr(), "\n";
clearstatcache();


function tests_performance_isPrimeSquarerootWayTwoArr(){
    global $iterations;
    global $isPrimeSquarerootWayTwo;
    $isPrimeSquarerootWayTwoArr = [];
    $isPrimeSquarerootWayTwoSum = 0;
    for ($i = 1; $i <= $iterations; $i++){
        $start = microtime(true);
        isPrimeSquarerootWayTwo(30000239);
        $end = microtime(true);
        // array_push($isPrimeSquarerootWayTwoArr, $time);
        $isPrimeSquarerootWayTwoSum = $isPrimeSquarerootWayTwoSum + ($end - $start);
    }
    return $isPrimeSquarerootWayTwoSum;
}

// $result = calculateAverage($isPrimeSquarerootWayTwoArr);
// echo "isPrimeSquarerootWayTwoArr: ", $result, "\n";
// $isPrimeSquarerootWayTwoArr = [];
echo "isPrimeSquarerootWayTwoSum: ", tests_performance_isPrimeSquarerootWayTwoArr(), "\n";
clearstatcache();


function tests_performance_primeArr(){
    global $iterations;
    global $prime;
    $primeArr = [];
    $primeSum = 0;
    for ($i = 1; $i <= $iterations; $i++){
        $start = microtime(true);
        prime(30000239);
        $end = microtime(true);
        // array_push($primeArr, $time);
        $primeSum = $primeSum + ($end - $start);
    }
    return $primeSum;
}

// $result = calculateAverage($primeArr);
// echo "primeArr: ", $result, "\n";
// $primeArr = [];
echo "primeSum: ", tests_performance_primeArr(), "\n";
clearstatcache();

